### ISSUE TITLE
[Bug]: Perform manual release overriding filters on page select all

### ISSUE THREAD

**Issue by danilovetatek** [microsoft/ALAppExtensions#26132]
### Describe the issue

When doing select all on a filtered page of Purchase/Sales Invoices/Orders/Cr.Memos, the Status filter is being overwritten by Microsoft's code that's trying to count total records. This is causing the action to affect non-selected lines.

The message also displays a negative number for items skipped due to this problem. 

### Expected behavior

Expected behavior is that the counter should apply the "<> Released" Status filter on top of the existing filters, to avoid this problem. This can be simply accomplished by marking the records.

### Steps to reproduce

From the Sales/Purchase Invoice or Purchase Credit Memo screens.  

On any line, click on the list of options at the line level and select "SELECT MORE".
Set the filter for "Status". (in the provided example, I set the status = Open)
Click on the Select All indicator in the header.
Choose any action to take on all lines.  In this example I've chosen to change all Open items to Released.
The message will tell me it will Release X items, and show a negative number for how many I skipped.

![image](https://github.com/microsoft/ALAppExtensions/assets/123086474/f815f544-3e51-41f1-81b7-b2d1de204792)

This is due to the code overriding my initial filter to count the records.

```
    internal procedure PerformManualRelease(var PurchaseHeader: Record "Purchase Header")
    var
        BatchProcessingMgt: Codeunit "Batch Processing Mgt.";
        NoOfSelected: Integer;
        NoOfSkipped: Integer;
    begin
        NoOfSelected := PurchaseHeader.Count();
        PurchaseHeader.SetFilter(Status, '<>%1', PurchaseHeader.Status::Released);
        NoOfSkipped := NoOfSelected - PurchaseHeader.Count;
        BatchProcessingMgt.BatchProcess(PurchaseHeader, Codeunit::"Purchase Manual Release", Enum::"Error Handling Options"::"Show Error", NoOfSelected, NoOfSkipped);
    end;
```

### Additional context

This should be fixed by either marking the records; or using a local variable to count the record which won't affect the initial filters selected on the page.

### I will provide a fix for a bug

- [X] I will provide a fix for a bug

**Comment by JesperSchulz** [comment]
@danilovetatek, this seems related to the BaseApp. Would you like me to move the bug to the [BaseApp contribution pilot](https://github.com/microsoft/BusinessCentralApps), where we can triage this issue?

**Comment by danilovetatek** [comment]
@JesperShulz this is correct. I would appreciate that. Could you please tell me how can I track this issue if you do so? I don't have access to the link you've attached above.

**Comment by aholstrup1** [comment]
If you fill out [this form](https://forms.office.com/pages/responsepage.aspx?id=v4j5cvGGr0GRqy180BHbR_Qj5hjzNeNOhBcvBoRIOltUOVBVTklZN1hBOTZJUU40OE5CUzNWNk1FQy4u) we can grant you access :)
